src/__tests__/ExpoClient-test.ts

```diff
@@ -187,6 +172,26 @@ describe('sending push notification messages', () => {
       others: expect.arrayContaining([expect.any(Error)]),
     });
   });
+
+  test('handles 429 too many request by applying exponential backoff', async () => {
+    (fetch as any).mock(
+      'https://exp.host/--/api/v2/push/send',
+      {
+        status: 429,
+        body: {
+          errors: [{ code: 'RATE_LIMIT_ERROR', message: `Rate limit exceeded` }],
+        },
+      },
+      { repeat: 3 }
+    );
+
+    const client = new ExpoClient();
+    const rejection = expect(client.sendPushNotificationsAsync([])).rejects;
+    await rejection.toThrowError(`Rate limit exceeded`);
+    await rejection.toMatchObject({ code: 'RATE_LIMIT_ERROR' });
+
+    expect((fetch as any).done()).toBeTruthy();
+  }, 10000);
```

Comment:
This is fairly tough to do due to the nature of `promiseRetry`. The actual first fetch call isn't made until the await, and the first `setTimeout` is added after the first fetch call fails. This makes it impossible as far as I can tell to call `advanceTimersByTime` (or any other fake timer methods) in the sequence necessary to execute the function.
```
const fnAsync = async () = {
   let res = await fetch();
   if (res.status = 429) {
      await setTimeout(1000);
      res = await fetch();
   }
   return res;
}
```
in test, neither of the following two options work:
no await:
```
fnAsync();
jest.advanceTimersByTime(1000); // this will do nothing since the first `fetch` line hasn't resolved yet (no await)
```

await:
```
await fnAsync();
jest.advanceTimersByTime(1000); // this will never be called (until fnAsync resolves and the execution proceeds)
```

Reply:
Sometimes you need to put in `await Promise.resolve()` to run another round of microtasks:

```
fnAsync();
await Promise.resolve();
jest.advanceTimersByTime(1000);
await Promise.resolve();
```